File: PYTHON/02-VARIABLES-OPERADORES/cociente_residuo.py
'''
Práctica01:
Calcular el conciente y residuo enunciado: Hallar el cociente y residuo (resto) de dos números enteros.
Análisis: Para la solución de este problema, se requiere que el usario ingrese dos nómeros enteros y el sistema realice el cálculo respectivo para hallar el cociente y residuo.
'''


print('Práctica de Cociente y Residuo de dos números. \nDebes Ingresar Dos números enteros:  ')

# Entrada de datos
a = input('Ingrese el Primer Número: ')
b = input('Ingrese el Segundo Número: ')

# Casting de datos para convertirlos en números antes de la operación
a = int(a)
b = int(b)

# Operación
cociente = a / b
residuo = a % b

# Imprimir en consola el resultado de la operación
resultado = f'Para los números ingresados, el resultado obtenido es: \nCociente: {cociente} \nResiduo: {residuo}'
print(resultado)

File: GO/src/golang/practicas/practica02/cocienteResiduo.go
package main

import "fmt"

/*
Practica 02: Calcular cociente y el Residuo de dos números enteros
Enunciado: halar el cociente y el residuo (resto) de dos números enteros.

Análisis: para la solución de este problema, se requiere que el usuario ingrese dos números enteros y el sistema realice el cálculo respectivo para hallar el cociente y residuo.
*/
func cociente() int {
	var a, b int

	// Ingreso de datos
	fmt.Print("\n( COCIENTE ) -> Ingrese un número ENTERO a continuación : ")
	fmt.Scanln(&a)
	fmt.Print("( COCIENTE ) -> Ingrese otro número ENTERO a continuación : ")
	fmt.Scanln(&b)

	// Calcular el cociente
	return a / b
}

func residuo() int {

	var a, b int

	// Ingreso de datos
	fmt.Print("\n\n( RESIDUO ) -> Ingrese un número ENTERO a continuación : ")
	fmt.Scanln(&a)
	fmt.Print("( RESIDUO ) -> Ingrese otro número ENTERO a continuación : ")
	fmt.Scanln(&b)

	// Calcular el residuo
	return a % b
}

func main() {
	c := cociente()
	fmt.Println("\nEl Cociente de los números que ingresó es:", c)

	r := residuo()
	fmt.Println("\nEl residuo de los números que ingresó es:", r)

}
